mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -110,27 +112,37 @@
         @Override
         public void onFailure(Call call, int statusCode, Headers headers, byte[] responseBody, Throwable error) {
             setProgressIndicatorVisible(false);
-            if (statusCode == 401) {
-                showAlertDialog(getString(R.string.error_authentication_failed));
-                return;
-            }
-            if (error instanceof java.net.UnknownHostException) {
+            Log.e(TAG, "Error: " + error.toString());
+            Log.e(TAG, "Code: " + statusCode);
+            if (statusCode >= 400){
+                int resourceID;
+                try {
+                    resourceID = getResources().getIdentifier("error_http_code_" + statusCode, "string", getPackageName());
+                    showMessageToUser(getString(resourceID), "dialog", 5);
+                } catch (android.content.res.Resources.NotFoundException e) {
+                    showMessageToUser(String.format(getString(R.string.error_http_connection_failed), statusCode), "dialog", 5);
+                }
+            } else if (error instanceof java.net.UnknownHostException) {
                 Log.e(TAG, "Unable to resolve hostname");
-                showError(error.getMessage());
+                showMessageToUser(getString(R.string.error_unable_to_resolve_hostname), "dialog", 5);
             } else if (error instanceof SSLHandshakeException) {
-                showError(getString(R.string.error_connection_sslhandshake_failed));
+                // if ssl exception, check for some common problems
+                if (error.toString().contains("java.security.cert.CertPathValidatorException")) {
```

Comment:
Wouldn't `error.getCause() instanceof CertPathValidatorException` be a better check?

Reply:
Yes, it is. Thanks.